lib/src/shared/main/java/com/couchbase/lite/Document.java

```diff
@@ -523,19 +535,6 @@ final FLSliceResult encode() throws LiteCoreException {
     // Private access
     //---------------------------------------------
 
-    private void free() {
-        root = null;
-
-        final C4Document c4Doc;
-        synchronized (lock) {
-            c4Doc = c4Document;
-            c4Document = null;
-        }
-
-        // c4doc should be retained.
-        if (c4Doc != null) { c4Doc.release(); }
```

Comment:
This is getting a little messy when retain and release are not matched. In updatedC4DocumentLocked(C4Document) we retain the c4Doc but not there is no where to call the release(). I tried to search where the retain() and release() method are called but I couldn't see if it's necessary to call retain()/release() at all as the C4Document will be released by itself. 

@bmeike Have you tried to clean this up, maybe by removing retain()/release() calls?

Reply:
I have tried to clean it up... but not in a single commit.  This is already ugly.
Let look at this again.  I may be able to just remove them all, altogether, as you suggest.